### PR TITLE
Added a new DOM Async strategy with React transpiling

### PR DESCRIPTION
**Pull request by Innos**
Reverted DOM strategy back and created a new strategy for Async DOM tests with ReactJS transpiling.

Added Babel-Core to the App.config and a new SubmissionType for the new Strategy

### CODE REVIEW

File: Open Judge System/Data/OJS.Data/Migrations/DefaultMigrationConfiguration.cs [hunk 1]
```diff
@@ -290,6 +291,18 @@ protected void SeedSubmissionTypes(OjsDbContext context)
                         AllowedFileExtensions = null,
```
Comment (YanaSlavcheva) [inline Open Judge System/Data/OJS.Data/Migrations/DefaultMigrationConfiguration.cs]: shift + tab

File: Open Judge System/Data/OJS.Data/Migrations/DefaultMigrationConfiguration.cs [hunk 2]
```diff
@@ -290,6 +291,18 @@ protected void SeedSubmissionTypes(OjsDbContext context)
...
                         AllowBinaryFilesUpload = false,
                     },
+                new SubmissionType
+                    {
```
Comment (YanaSlavcheva) [inline Open Judge System/Data/OJS.Data/Migrations/DefaultMigrationConfiguration.cs]: shift + tab

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy.cs [hunk 3]
```diff
@@ -0,0 +1,236 @@
+﻿namespace OJS.Workers.ExecutionStrategies
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text.RegularExpressions;
+    using OJS.Workers.Common;
+
+    public class NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy : NodeJsPreprocessExecuteAndRunJsDomUnitTestsExecutionStrategy
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy.cs]: split on two rows pls

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy.cs [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+            string sinonModulePath,
+            string sinonChaiModulePath,
+            string underscoreModulePath,
+            string babelCoreModulePath,
+            string reactJsxPluginPath,
+            string reactModulePath,
+            string reactDomModulePath,
+            int baseTimeUsed,
+            int baseMemoryUsed)
+            : base(
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy.cs]: tab

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy.cs [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+            nodeJsExecutablePath,
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy.cs]: tab this row + the rows below

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy.cs (hunk 4 above)
Comment (Innos) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy.cs]: Should the call to base constructor be tabbed?

Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsExecuteAndRunAsyncJsDomTestsWithReactExecutionStrategy.cs]: It is fine now, sorry :)

**Pull request merged**